events-api/src/modules/events/events.repository.ts

```diff
@@ -18,4 +21,107 @@ export class EventsRepository {
 
     return result;
   }
+
+  static async create(event: CreateEventDto): Promise<EventModel> {
+    try {
+      const result: EventModel[] = await db.execute<EventModel>(
+        `INSERT INTO events (event_id, event_name, event_description, event_location, event_date)
+         VALUES (?, ?, ?, ?, ?)
+         RETURNING event_id as eventId,event_name as eventName, event_description as eventDescription,
+         event_location as eventLocation,event_date as eventDate,event_created_at as eventCreatedAt, event_updated_at as eventUpdatedAt;`,
+        [
+          faker.string.uuid(),
+          event.eventName,
+          event.eventDescription,
+          event.eventLocation,
+          new Date(event.eventDate)
+            .toISOString()
+            .replace("T", " ")
+            .replace("Z", ""),
+          ,
+        ],
+      );
+
+      console.log("Insert new event result: ", result);
+      return result[0]!;
+    } catch (err) {
+      console.error("EventRepositoryError: couldn't create event!", err);
+      throw new Error();
+    }
+  }
+
+  static async update(
+    eventId: string,
+    updatedEvent: UpdateEventDto,
+  ): Promise<string> {
+    try {
+      const event = await EventsRepository.getById(eventId);
+      await db.execute<EventModel>(
+        `UPDATE events
+         SET
+            event_name = ?,
+            event_description = ?,
+            event_location = ?,
+            event_date = ?
+         WHERE
+            event_id = ?;`,
+        [
+          updatedEvent.eventName || event?.eventName,
+          updatedEvent.eventDescription || event?.eventDescription,
+          updatedEvent.eventLocation || event?.eventLocation,
+          new Date(updatedEvent.eventDate || event?.eventDate || "")
```

Comment:
Why to have a fallback for empty string?

Reply:
Honestly, I couldn't find other solutions after the compiler complained I had to. It'd be great if you could  share possible other solution....